### ISSUE TITLE
Can I use MP4Box to split one video into multiple parts and merge them into one video?

### ISSUE THREAD

**Issue by zydjohnHotmail** [gpac/gpac#2186]
Thanks for reporting your issue. Please make sure these boxes are checked before submitting your issue - thank you!

- [X] I looked for a similar issue and couldn't find any.
- [X] I tried with the latest version of GPAC. Installers available at http://gpac.io/downloads/gpac-nightly-builds/
- [X] I give enough information for contributors to reproduce my issue (meaningful title, github labels, platform and compiler, command-line ...). I can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

Detailed guidelines: http://gpac.io/2013/07/16/how-to-file-a-bug-properly/
Hello:
I have one video (MP4 format) file: 0.mp4.
I want to split it by 3 time ranges: from 00:00:10 to 00:00:20; from 00:00:45 to 00:01:00; from 00:02:00 to 00:03:00.  And then merge/concat them into one video file: merge.mp4.
I know I can use splitx to split 0.mp4 into 3 files, like: 1.mp4, 2.mp4, 3.mp4, then run -force-cat command to merge/concat them into one file: merge.mp4
But I want to know if I can use one command to skip the middle steps and generate the result directly.
FFMPEG has similar command like: between(t, t0, t1) + between(t, t2, t3) ...
But I want to know if I can do the same with MP4box.
Thanks,

**Comment by rbouqueau** [comment]
Thanks for reporting. It seems it doesn't work yet but we are really not far.

@jeanlf Is there a reason for applying a specific parsing in MP4Box instead of forwarding to the reframer filter?

```
$ MP4Box -splitx 0-1,0-1 tmp.mp4 -out tmp2.mp4 -graph
splitting: file 1 done

Filters connected:
mp4dmx (mov=0x556ac4a94360:alltk) (idx=1)
-(PID PID1) reframer (splitrange:xround=closest:probe_ref:xs=0:xe=1,0-1) (idx=2)
--(PID PID1) mp4mx (dyn_idx=4)
---(PID PID1) fout (dst=tmp2.mp4) (idx=3)
```

https://github.com/gpac/gpac/wiki/mp4box-dump-opts#file-splitting

https://github.com/gpac/gpac/wiki/reframer#range-extraction

**Comment by jeanlf** [comment]
> Is there a reason for applying a specific parsing in MP4Box instead of forwarding to the reframer filter?

Yes for backward compatibility reasons when moving to filters, we did not change -splitx (single chunk extraction).  

We can look into that, but meanwhile I suggest you use gpac and the reframer filter directly (here with times in seconds):
```
gpac -i source reframer:xs=10,45,120:xe=20,60,180 -o cur.mp4
```

**Comment by jeanlf** [comment]
now on master using comma-separated list - also added delta time for end specifier using `D`:

 ```
MP4Box -splitx 10-D10,45-D15,120-180 source.mp4 -o split.mp4

```

**Comment by jeanlf** [comment]
Sorry typo, should be `-out` not `-o-`
